Calibration/EcalAlCaRecoProducers/src/AlCaElectronsTest.h

```diff
@@ -23,266 +26,171 @@
-  m_barrelLocalCrystalsEnergy = new TH2F ("m_barrelLocalCrystalsEnergy","m_barrelLocalCrystalsEnergy",20,-10,10,20,-10,10) ;
-  m_endcapGlobalCrystalsEnergy = new TH2F ("m_endcapGlobalCrystalsEnergy","m_endcapGlobalCrystalsEnergy",100,0,100,100,0,100) ;
-  m_endcapLocalCrystalsEnergy = new TH2F ("m_endcapLocalCrystalsEnergy","m_endcapLocalCrystalsEnergy",20,-10,10,20,-10,10) ;
-  m_barrelGlobalCrystalsEnergyMap = new TH2F ("m_barrelGlobalCrystalsEnergyMap","m_barrelGlobalCrystalsEnergyMap",171,-85,86,360,0,360) ;
-  m_endcapGlobalCrystalsEnergyMap = new TH2F ("m_endcapGlobalCrystalsEnergyMap","m_endcapGlobalCrystalsEnergyMap",100,0,100,100,0,100) ;
-   return ;
+void AlCaElectronsTest::beginJob() {
+  edm::Service<TFileService> fs;
+  m_barrelGlobalCrystalsMap =
+      fs->make<TH2F>("m_barrelGlobalCrystalsMap", "m_barrelGlobalCrystalsMap", 171, -85, 86, 360, 0, 360);
+  m_barrelLocalCrystalsMap =
+      fs->make<TH2F>("m_barrelLocalCrystalsMap", "m_barrelLocalCrystalsMap", 20, -10, 10, 20, -10, 10);
+  m_endcapGlobalCrystalsMap =
+      fs->make<TH2F>("m_endcapGlobalCrystalsMap", "m_endcapGlobalCrystalsMap", 100, 0, 100, 100, 0, 100);
+  m_endcapLocalCrystalsMap =
+      fs->make<TH2F>("m_endcapLocalCrystalsMap", "m_endcapLocalCrystalsMap", 20, -10, 10, 20, -10, 10);
+  m_barrelGlobalCrystalsEnergy =
+      fs->make<TH2F>("m_barrelGlobalCrystalsEnergy", "m_barrelGlobalCrystalsEnergy", 171, -85, 86, 360, 0, 360);
+  m_barrelLocalCrystalsEnergy =
+      fs->make<TH2F>("m_barrelLocalCrystalsEnergy", "m_barrelLocalCrystalsEnergy", 20, -10, 10, 20, -10, 10);
+  m_endcapGlobalCrystalsEnergy =
+      fs->make<TH2F>("m_endcapGlobalCrystalsEnergy", "m_endcapGlobalCrystalsEnergy", 100, 0, 100, 100, 0, 100);
+  m_endcapLocalCrystalsEnergy =
+      fs->make<TH2F>("m_endcapLocalCrystalsEnergy", "m_endcapLocalCrystalsEnergy", 20, -10, 10, 20, -10, 10);
+  m_barrelGlobalCrystalsEnergyMap =
+      fs->make<TH2F>("m_barrelGlobalCrystalsEnergyMap", "m_barrelGlobalCrystalsEnergyMap", 171, -85, 86, 360, 0, 360);
+  m_endcapGlobalCrystalsEnergyMap =
+      fs->make<TH2F>("m_endcapGlobalCrystalsEnergyMap", "m_endcapGlobalCrystalsEnergyMap", 100, 0, 100, 100, 0, 100);
 }
 
-
 // ----------------------------------------------------------------
 
-
-void 
-AlCaElectronsTest::endJob ()
-{      
-   TFile output (m_outputFileName.c_str (),"recreate") ;
-   m_barrelGlobalCrystalsMap->Write () ;
-   m_barrelLocalCrystalsMap->Write () ;
-   m_endcapGlobalCrystalsMap->Write () ;
-   m_endcapLocalCrystalsMap->Write () ;   
-   m_barrelGlobalCrystalsEnergy->Write () ;
-   m_barrelLocalCrystalsEnergy->Write () ;
-   m_endcapGlobalCrystalsEnergy->Write () ;
-   m_endcapLocalCrystalsEnergy->Write () ;   
-   m_barrelGlobalCrystalsEnergyMap->Write () ;
-   m_endcapGlobalCrystalsEnergyMap->Write () ;
-   output.Close () ;
-   //PG save root things
-   return ;
-}
-
+void AlCaElectronsTest::endJob() {}
```

Comment:
If this is not doing anything, could be removed, right?

Reply:
Done